src/FSharpPlus/Extensions/Validation.fs

```diff
@@ -0,0 +1,80 @@
+﻿namespace FSharpPlus
+
+[<RequireQualifiedAccess>]
+module Validations =
+    open System
+    open FSharpPlus.Data
+    
+    let inline validate errorMessage f v =
+        if f v then
+            Success v
+        else
+            Failure [ errorMessage ]
+
+    let inline requireString propName =
+        let errorMessage =
+            sprintf "%s cannot be null, empty or whitespace." propName
```

Comment:
In my experience using string as errors doesn't scale well, unless they are at the boundaries with the user.
But otherwise, validations errors should be represented with a custom type, normally a DU, which should be directly related to the business model of validation errors.

Having said that, we could add an additional parameter which is a function that creates the error, so the user is free to supply whatever structure he uses.

Reply:
@rodriguestiago0 
1. Error strings must be declared as public static functions to be available for plain reuse (for example in unit tests)
2. We have a case right now when we need to delay error string creation till it is required:
    a. Output to log as is
    b. Output to API response with camel case field name modification

Reply:
I know to let the user define the Error to be added. It can be a string, int or whatever the developer wants.